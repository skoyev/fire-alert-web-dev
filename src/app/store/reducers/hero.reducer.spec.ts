import { Hero } from '@appModels/hero';
import {
  AddHero,
  AddHeroSuccess,
  DeleteHero,
  DeleteHeroSuccess,
  GetHeroById,
  GetHeroByIdSuccess,
  GetHeroes,
  GetHeroesSuccess,
  HeroError,
  SearchHeroes,
  SearchHeroesReset,
  SearchHeroesSuccess,
  UpdateHero,
  UpdateHeroSuccess
} from '@appStore/actions/hero.actions';
import * as fromReducer from './hero.reducer';

describe('hero reducer', () => {
  const init = { type: '@@init' } as any;
  let mockHero: Hero;
  let initialState: fromReducer.State;

  beforeEach(() => {
    mockHero = { id: 2, name: 'hero2' };

    initialState = fromReducer.initialState;
  });

  describe('State Changes', () => {
    it('should have an initial state', () => {
      const state = fromReducer.reducer(initialState, init);

      expect(state).toEqual(initialState);
    });

    it('should get heroes', () => {
      const getHeroes = new GetHeroes();
      const state = [init, getHeroes].reduce(fromReducer.reducer, initialState);

      expect(state).toMatchSnapshot();
    });

    it('should get heroes success', () => {
      const getHeroesSuccess = new GetHeroesSuccess([
        { id: 0, name: 'hero0' },
        { id: 1, name: 'hero1' }
      ]);
      const state = [init, getHeroesSuccess].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should add hero', () => {
      const addHero = new AddHero(mockHero);
      const state = [init, addHero].reduce(fromReducer.reducer, initialState);

      expect(state).toMatchSnapshot();
    });

    it('should add hero success', () => {
      const addHeroSuccess = new AddHeroSuccess(mockHero);
      const state = [init, addHeroSuccess].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should get hero by id', () => {
      const getHeroById = new GetHeroById(0);
      const state = [init, getHeroById].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should get hero by id success', () => {
      const getHeroByIdSuccess = new GetHeroByIdSuccess(mockHero);
      const state = [init, getHeroByIdSuccess].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should update hero', () => {
      const updateHero = new UpdateHero({ id: 0, name: 'new hero name' });
      const state = [init, updateHero].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should update hero success', () => {
      const updateHeroSuccess = new UpdateHeroSuccess({
        id: 0,
        name: 'new hero name'
      });
      const state = [init, updateHeroSuccess].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should delete hero', () => {
      const deleteHero = new DeleteHero({ id: 0 } as Hero);
      const state = [init, deleteHero].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should delete hero success', () => {
      const getHeroesSuccess = new GetHeroesSuccess([
        { id: 0, name: 'hero0' },
        { id: 1, name: 'hero1' }
      ]);
      const deleteHeroSuccess = new DeleteHeroSuccess({ id: 0 } as Hero);
      const state = [init, getHeroesSuccess, deleteHeroSuccess].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should search heroes', () => {
      const searchHeroes = new SearchHeroes('hero');
      const state = [init, searchHeroes].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should search heroes success', () => {
      const searchHeroesSuccess = new SearchHeroesSuccess([
        { id: 0, name: 'hero0' },
        { id: 1, name: 'hero1' }
      ]);
      const state = [init, searchHeroesSuccess].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should search heroes reset', () => {
      const searchHeroesReset = new SearchHeroesReset();
      const state = [init, searchHeroesReset].reduce(
        fromReducer.reducer,
        initialState
      );

      expect(state).toMatchSnapshot();
    });

    it('should hero error', () => {
      const heroError = new HeroError('error');
      const state = [init, heroError].reduce(fromReducer.reducer, initialState);

      expect(state).toMatchSnapshot();
    });
  });
});
