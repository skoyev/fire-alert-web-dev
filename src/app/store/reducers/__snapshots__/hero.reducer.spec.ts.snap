// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hero reducer State Changes should add hero 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": true,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should add hero success 1`] = `
Object {
  "entities": Object {
    "2": Object {
      "id": 2,
      "name": "hero2",
    },
  },
  "error": null,
  "ids": Array [
    2,
  ],
  "loaded": true,
  "loading": false,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should delete hero 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": true,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should delete hero success 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "id": 1,
      "name": "hero1",
    },
  },
  "error": null,
  "ids": Array [
    1,
  ],
  "loaded": true,
  "loading": false,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should get hero by id 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": true,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should get hero by id success 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": 2,
}
`;

exports[`hero reducer State Changes should get heroes 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": true,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should get heroes success 1`] = `
Object {
  "entities": Object {
    "0": Object {
      "id": 0,
      "name": "hero0",
    },
    "1": Object {
      "id": 1,
      "name": "hero1",
    },
  },
  "error": null,
  "ids": Array [
    0,
    1,
  ],
  "loaded": true,
  "loading": false,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should hero error 1`] = `
Object {
  "entities": Object {},
  "error": "error",
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should search heroes 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": true,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should search heroes reset 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should search heroes success 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": false,
  "searchHeroes": Array [
    Object {
      "id": 0,
      "name": "hero0",
    },
    Object {
      "id": 1,
      "name": "hero1",
    },
  ],
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should update hero 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": false,
  "loading": true,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;

exports[`hero reducer State Changes should update hero success 1`] = `
Object {
  "entities": Object {},
  "error": null,
  "ids": Array [],
  "loaded": true,
  "loading": false,
  "searchHeroes": null,
  "searchTerm": "",
  "selectedHeroId": null,
}
`;
